Hi @PaulYuuu, the PR that we thought is a blocker for this is merged but I still get some problems here like
```
Run python setup.py bdist_egg
  python setup.py bdist_egg
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.12.11/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.12.11/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.11/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.11/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.11/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.12.11/x64/lib
Traceback (most recent call last):
  File "/home/runner/work/avocado-vt/avocado-vt/setup.py", line 17, in <module>
    from setuptools import find_packages, setup
ModuleNotFoundError: No module named 'setuptools'
Error: Process completed with exit code 1.
```
So I assume we have to deprecate the setuptools as well.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -98,9 +98,10 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install build setuptools
+        pip install build setuptools wheel
     - name: Build package
       run: python -m build
-    - name: Build eggs
+    - name: Build eggs (legacy)
+      if: ${{ contains(fromJSON('["3.9","3.10","3.11"]'), matrix.python-version) }}
       run: python setup.py bdist_egg
     - run: echo "🥑 This job's status is ${{ job.status }}."

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -35,7 +35,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
       fail-fast: false
 
     steps:
@@ -82,7 +82,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
       fail-fast: false
 
     steps:
@@ -94,7 +94,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install build
+        pip install build setuptools
     - name: Build package
       run: python -m build
     - name: Build eggs

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -35,7 +35,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: &py_versions ["3.9", "3.10", "3.11", "3.12", "3.13"]
       fail-fast: false
 
     steps:
@@ -82,7 +82,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: *py_versions
       fail-fast: false
 
     steps:

